src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEReader.cs

```diff
@@ -690,11 +690,13 @@ public bool TryOpenAssociatedPortablePdb(string peImagePath, Func<string, Stream
 
             // First try .pdb file specified in CodeView data (we prefer .pdb file on disk over embedded PDB
             // since embedded PDB needs decompression which is less efficient than memory-mapping the file).
-            var codeViewEntry = entries.FirstOrDefault(e => e.IsPortableCodeView);
-            if (codeViewEntry.DataSize != 0 && 
-                TryOpenCodeViewPortablePdb(codeViewEntry, peImageDirectory, pdbFileStreamProvider, out pdbReaderProvider, out pdbPath, ref errorToReport))
+            foreach (var codeViewEntry in entries)
             {
-                return true;
+                if (codeViewEntry.DataSize != 0 && 
+                    TryOpenCodeViewPortablePdb(codeViewEntry, peImageDirectory, pdbFileStreamProvider, out pdbReaderProvider, out pdbPath, ref errorToReport))
```

Comment:
A lot of tests are failing with this change because of you are trying to pass non-portable .ni. pdbs into TryOpenCodeViewPortablePdb now.